### PR TITLE
User homepage and SearchPage UI

### PR DESCRIPTION
**Pull request by SemanBoon**
## What type of PR is this?

- [X] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR introduces some visual enhancements to the SheBraids app:

1. User Home Screen: Added a custom tooltip on hover for the category sections and applied a gradient background for a fresh look. These changes include some hover effects to make the experience more interactive.
2. NavBar Component: The NavBar now has a clean, minimal design with an animated underline effect on hover for each menu item.
3. Search Page: The layout now features a large "Search" title at the top, with an input field and buttons neatly aligned below. The back button has been styled to be simple text, placed at the top left corner for easy navigation.

## Screenshots, Recordings
Screenshot 2024-07-30 at 11 23 36 AM

Screenshot 2024-07-30 at 11 24 01 AM

## Test Plan

 I manually checked the new features on the User Home Screen, NavBar, and Search Page to ensure everything looks great and works smoothly.

## What challenges, debugging steps, and learnings did you encounter with this PR?
This PR was a great opportunity for me to dive deeper into CSS animations and understand how they can enhance user experience, especially when ensuring a responsive design.

### CODE REVIEW

File: Backend/index.js [hunk 1]
```diff
@@ -620,14 +620,17 @@ app.post('/get-available-slots', async (req, res) => {
...
+        where: { id: providerId },
+        select: { schedule: true },
       });
       if (!provider) {
-              return res.status(404).json({ error: 'Service provider not found' });
+        return res.status(404).json({ error: 'Service provider not found' });
       }
-      const availableSlots = getAvailableSlots(provider.schedule, serviceDuration, {}); // Slot popularity passed as {}
+      const availableSlots = getAvailableSlots(provider.schedule, serviceDuration, {});
+      console.log(provider.schedule)
```
Comment (anykac) [inline Backend/index.js]: are these console logs supposed to stay here?

File: Backend/index.js [hunk 2]
```diff
@@ -641,14 +642,14 @@ app.post('/get-recommended-slots', async (req, res) => {
       const provider = await prisma.serviceProvider.findUnique({ where: { id: providerId } });
 
       if (!user || !provider) {
-          return res.status(404).json({ error: 'User or Service Provider not found' });
+        return res.status(404).json({ error: 'User or Service Provider not found' });
```
Comment (karen-ka) [inline Backend/index.js:645]: Are we using a new auto formatter? Just wondering b/c these format changes seem unrelated

File: Backend/utils.js [hunk 3]
```diff
@@ -38,24 +38,21 @@ const getUpdatedWeights = (priority) => {
 const getAvailableSlots = (schedule, serviceDuration, slotPopularity) => {
     const availableSlots = [];
     const requiredSlots = Math.ceil(serviceDuration * 60 / 30);
-    // service.duration unit is hours.
-    // For 30 mins, it will be 0.5 * 60/ 30 = 1, in this case required slots is 1
-    // For 2 hours, it will be 2 * 60/30 = 4, in this case required slots is 4
```
Comment (anykac) [inline Backend/utils.js:43]: why dont we keep these comments?

File: frontend/src/Components/UserHomePage/UserHomePage.css [hunk 4]
```diff
@@ -1,81 +1,130 @@
...
-.braid-section:hover {
-  background-color: #e0e0e0;
+.category-section:hover .tooltip {
+  opacity: 1;
+  visibility: visible;
 }
-.weave-section:hover {
-  background-color: #e0e0e0;
+
+.appointment-card {
```
Comment (karen-ka) [inline frontend/src/Components/UserHomePage/UserHomePage.css:92]: Nice job putting these all in the related home page css file

File: frontend/src/Components/UserHomePage/UserHomePage.jsx [hunk 5]
```diff
@@ -47,22 +47,36 @@ const UserHomePage = () => {
...
         <h1>Welcome to the SheBraids, {user.name}!</h1>
         <h3>Find Your Perfect Match</h3>
       </div>
-      <div className = "container">
-        <div className="braid-section"onClick={() => handleCategoryClick('Braids')}>BRAIDS</div>
-        <div className="haircut-section"onClick={() => handleCategoryClick('Haircuts')}>HAIRCUTS</div>
-        <div className="weave-section"onClick={() => handleCategoryClick('Weave and Installs')}>WEAVE AND INSTALLS</div>
-        <div className="locs-section"onClick={() => handleCategoryClick('Locs')}>LOCS</div>
+      <div className = "home-container">
+      <div className="category-section braid-section" onClick={() => handleCategoryClick('Braids')}>
```
Comment (karen-ka) [inline frontend/src/Components/UserHomePage/UserHomePage.jsx:57]: good usage of classes!

**Review by karen-ka — APPROVED** [review]
love the style update!
